app/uk/gov/hmrc/economiccrimelevyregistration/navigation/IsUkAddressPageNavigator.scala

```diff
@@ -36,10 +36,18 @@ class IsUkAddressPageNavigator @Inject() (
     registration: Registration
   )(implicit request: RequestHeader): Future[Call] =
     registration.contactAddressIsUk match {
-      case Some(ukMode) => addressLookupFrontendConnector.initJourney(ukMode).map(journeyUrl => Call(GET, journeyUrl))
+      case Some(ukMode) =>
+        addressLookupFrontendConnector.initJourney(ukMode, NormalMode).map(journeyUrl => Call(GET, journeyUrl))
       case _            => Future.successful(routes.JourneyRecoveryController.onPageLoad())
     }
 
-  override protected def navigateInCheckMode(registration: Registration): Future[Call] = ???
+  override protected def navigateInCheckMode(
+    registration: Registration
+  )(implicit request: RequestHeader): Future[Call] =
+    registration.contactAddressIsUk match {
```

Comment:
This is a duplicate of the normal mode navigation routing logic... move into a common function?